### ISSUE TITLE
Measuring curriculum effectiveness

### ISSUE THREAD

**Issue by oliverjam** [foundersandcoders/master-reference#988]
Do we have any way to quantify roughly how effective our curriculum is? Have we ever considered something like a short (maybe voluntary) end of week quiz to assess how well the week's learning outcomes were met?

It would be relatively easy to create given each week already has learning outcomes—do we think students would object to answering some informal questions (maybe just in a Google Form?)

**Comment by astroash** [comment]
@oliverjam when I was cf I creates google forms based off of each weeks learning objectives & asked students to fill them out. I used the feedback to give additional advice and partially inform my recommendations to changes in the curriculum. 

I personally found it useful but was cautious for it to become mandatory as back then the CF role was only part time. If there are others to review the feedback personally I think it could be a useful way to gather information, especially if its run as a short term initiative to revamp the curriculum.

here's an example of one:
https://forms.gle/8U1e3ePm8VbqZDE59

**Comment by oliverjam** [comment]
That's amazing, thanks. So you were just asking students if they felt like they knew each learning outcome? Rather than actually asking a question with a correct/wrong answer to check if they had learned it?

I think this would be useful data to gather after each week whether the CF or mentors used it or not. We'd be able to identify trends over the cohorts and figure out places the curriculum is lacking over time.

**Comment by astroash** [comment]
Yeah, there were whispers about changing the LOs back then so that was the main focus of the survey. If I did it again I would make the questions less binary than they appear here.  

Personally i prefer self assessment rather than quizes as they can promote competition, and anxiety in students who feel they are falling behind. How about a self assessment with prompting questions?

eg 
> How comfortable are you with types of HTTP requests
> - Whats the difference between a GET & POST request?
> - What type of request would you use for sending passwords

This way you're getting students to reflect on their current knowledge, and you're giving them questions that they can research/discuss with peers to fill in the gaps.

**Comment by oliverjam** [comment]
Yeah I like that a lot. I agree that I'm not comfortable putting even more stress on students who might be struggling with a quiz they might perceive as judging their progress. I guess anonymity should help a little, but self-assessment and reflection is a great way to do it.

Do we think it's too late to ask FAC16 to answer some surveys? I'm happy to go through the learning outcomes for each week and draw up a Google form

**Comment by charlielafosse** [comment]
I really like this idea.

I suppose that SGCs are meant to cover feedback, although they're a less than ideal format for people who are hesitant about public speaking. I also think that those who are less comfortable with a topic actually end up saying less in the SGC, which is definitely not helpful for getting a representative idea of how the week's gone. 

It might also be cool for students to be able to keep a record of their own survey responses over a cohort? And keep track of their progress and weak points etc. It'd be easy to submit the form, collect the data and send it back to students using Airtable (I have less experience with Google forms). Something seems very familiar here... yoohoo [FAC Track](https://github.com/fac-15/FAC-TRACK)
